vtex-apps/return-app
# V3 submit rma

This PR creates the backbone to connect the front end form to enter the RMA details to the back end to save it on mastedata. It brings validation for the fields in the form and placeholders for the required message. There a couple of missing pieces (loading, and error state, style and css handles, intl) but the code enables some parallel work on top of it.

It also allows us to test the orders that the user will be able to see when listing them to create a new rma. It will prevent the return of items already invoiced as input type (return invoices), and items on categories selected as forbidden to return in the admin settings. It DOES NOT remove items included in other RMA and not invoiced to return yet.

It should work for call center operators impersonating store users.

### How to test it?

Log in as store user in this [workspace](https://rmav3submitrma--powerplanet.myvtex.com). Go through the flow to create a new RMA (you can test the validation by not entering the required fields). When submitting the request, the UI will show the request id. Then, you can check it using the API:

`http://powerplanet.vtexcommercestable.com.br/api/dataentities/vtex_return_app_returnRequest/documents/{documentId}?_fields=_all`

The return body should match the values you submitted. The fields associated to the RMA value (totalReturnAmount and returnTotals) are not ready yet.
